ghost/link-tracking/lib/LinkClickTrackingService.js

```diff
@@ -62,6 +65,45 @@ class LinkClickTrackingService {
         this.#initialised = true;
     }
 
+    /**
+     * @param {object} options
+     * @param {string} options.filter
+     */
+    async getLinks(options) {
+        const data = [];
+
+        const postLinks = await this.#postLinkRepository.getAll({
+            filter: options.filter
+        });
+
+        if (postLinks.length === 0) {
+            return data;
+        }
+
+        const links = await this.#linkRedirectService.getAll({
+            filter: `id:[${postLinks.map(x => x.link_id.toHexString())}]`
+        });
+
+        for (const link of links) {
+            const events = await this.#linkClickRepository.getAll({
+                filter: `link_id:[${link.link_id.toHexString()}]`
+            });
+
+            const result = {
+                id: link.link_id.toHexString(),
+                url: link.to.toString(),
+                post_id: postLinks.find((postLink) => {
+                    return postLink.link_id.equals(link.link_id);
+                }).post_id.toHexString(),
+                click_events: events
+            };
+
+            data.push(result);
+        }
```

Comment:
IMO this is where the repository pattern is lacking. We have to do `this.#linkRedirectService.getAll` effectively is returning the same database models in the implementation. And we are fetching all the events of all the links in a loop. I'm not sure why we need to return all the event when we are only going to display the total count.